src/common/core/mixincore.js

```diff
@@ -0,0 +1,793 @@
+        };
+
+        this.getConstraintNames = function (node) {
+            return getValidNames(node, innerCore.getConstraintNames, []);
+        };
+
+        this.getJsonMeta = function (node) {
+            var meta = {children: {}, attributes: {}, pointers: {}, aspects: {}, constraints: {}},
+                nullRule = {items: [], minItems: [], maxItems: []},
+                tempNode,
+                names,
+                pointer,
+                i, j;
+
+            meta.children = self.getChildrenMeta(node);
+            if (meta.children) {
+                convertRuleToItemizedArraySet(meta.children);
+            } else {
+                meta.children = nullRule;
+            }
+
+            //attributes
+            names = self.getValidAttributeNames(node);
+            for (i = 0; i < names.length; i++) {
+                meta.attributes[names[i]] = self.getAttributeMeta(node, names[i]);
+            }
+
+            //pointers
+            names = self.getValidPointerNames(node);
+            for (i = 0; i < names.length; i++) {
+                meta.pointers[names[i]] = self.getPointerMeta(node, names[i]);
+                convertRuleToItemizedArraySet(meta.pointers[names[i]]);
+            }
+
+            //sets
+            names = self.getValidSetNames(node);
+            for (i = 0; i < names.length; i++) {
+                meta.pointers[names[i]] = self.getPointerMeta(node, names[i]);
+                convertRuleToItemizedArraySet(meta.pointers[names[i]]);
+            }
+
+            //aspects
+            names = self.getValidAspectNames(node);
+
+            for (i = 0; i < names.length; i++) {
+                meta.aspects[names[i]] = self.getAspectMeta(node, names[i]);
+            }
+
+            //constraints
+            names = self.getConstraintNames(node);
+            for (i = 0; i < names.length; i++) {
+                meta.constraints[names[i]] = self.getConstraint(node, names[i]);
+            }
+
+            return meta;
+        };
+
+        this.getOwnJsonMeta = function (node) {
+            // We have to extend the meta definition by the mixins, but only if we talk about the own rule-set.
+            var jsonMeta = innerCore.getOwnJsonMeta(node),
+                mixins = this.getMixinPaths(node);
+
+            if (mixins.length > 0) {
+                jsonMeta.mixins = mixins;
+            }
+
+            return jsonMeta;
+        };
+
+        this.getValidChildrenPaths = function (node) {
+            return getValidNames(node, innerCore.getValidChildrenPaths, []);
+        };
+
+        this.getChildrenMeta = function (node) {
+            var ruleHolders = getAllMatchingRuleHolders(node, 'containment', containmentGetter, []),
+                childrenMeta = {},
+                i;
+
+            for (i = 0; i < ruleHolders.length; i += 1) {
+                extendUniqueKeyedObject(childrenMeta, innerCore.getChildrenMeta(ruleHolders[i]));
```

Comment:
Is this function recursive? If it is not, we should not pass down a variable and change it in the function.